### PR TITLE
Github Code Scanning Integration

### PR DESCRIPTION
**Pull request by jeffersonking**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,6 +95,11 @@
                 "command": "sarif.clearState",
                 "category": "SARIF",
                 "title": "Clear State"
+            },
+            {
```
Comment (jeffersonking) [inline package.json:99]: Remove the Toggle.

File: src/extension/antiDrift.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Change } from 'diff';
+
+export function measureDrift(diffBlocks: Change[], offsetStart: number, offsetEnd: number): number | undefined {
```
Comment (jeffersonking) [inline src/extension/antiDrift.ts:6]: Filename?

File: src/extension/getOriginalDoc.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Uri } from 'vscode';
+import { antiDriftEnabled } from './antiDriftToggle';
+import { AnalysisInfo, getInitializedGitApi } from './index.activateGithubAnalyses';
+import { StringTextDocument } from './stringTextDocument';
+
+// TOOD: Consider caching the retval.
```
Comment (jeffersonking) [inline src/extension/getOriginalDoc.ts:9]: Spelling.

File: src/extension/index.activateGithubAnalyses.ts [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+        If not equal to default (true):
+        {
+            defaultValue: false
+            globalValue: true
+        }
+    */
+    const connectToGithubCodeScanning = workspace.getConfiguration('sarif-viewer').get<'off' | 'on' | 'onWithIntroduction'>('connectToGithubCodeScanning');
+    if (connectToGithubCodeScanning === 'off') return;
+
+    const config = {
```
Comment (jeffersonking) [inline src/extension/index.activateGithubAnalyses.ts:70]: Move into a more local scope?

File: src/extension/index.activateGithubAnalyses.ts [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+        user: '',
+        repoName: '',
+    };
+
+    // TODO: Block re-entrancy.
+    observe(store, 'remoteAnalysisInfoUdpated', () => updateAnalysisInfo());
```
Comment (jeffersonking) [inline src/extension/index.activateGithubAnalyses.ts:76]: Spelling.

File: src/extension/index.activateGithubAnalyses.ts (hunk 4 above)
Comment (jeffersonking) [reply src/extension/index.activateGithubAnalyses.ts:70]: This is the correct scope.

**Review by EasyRhinoMSFT — APPROVED** [review]

**Pull request merged**